service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ObservationStatementMapper.java

```diff
@@ -161,23 +166,26 @@ private String prepareComment(Observation observation) {
 
                 Optional.ofNullable(coding).ifPresent(code -> {
                     if (code.hasCode() || code.hasDisplay()) {
-                        commentBuilder.append("Interpretation Code: ");
+                        commentBuilder.append(INTERPRETATION_CODE_PREFIX);
                     }
                     if (code.hasCode()) {
                         commentBuilder.append(coding.getCode()).append(StringUtils.SPACE);
                     }
                     if (code.hasDisplay()) {
                         commentBuilder.append(code.getDisplay()).append(StringUtils.SPACE);
                     }
-
-                    if (interpretation.hasText()) {
-                        commentBuilder.append("Interpretation Text: ").append(interpretation.getText()).append(StringUtils.SPACE);
-                    }
                 });
             }
         }
 
-        commentBuilder.append(observation.hasComment() ? observation.getComment() : StringUtils.EMPTY);
+        String interpretationTextAndComment = Stream.concat(
+            Optional.ofNullable(observation.getInterpretation().getText()).stream(),
```

Comment:
Are we sure that interpretation is not null at this point? We're outside the `if (observation.hasInterpretation()) {...}` clause.

Reply:
Interestingly it doesn't fail. If there is no interpretation in FHIR, `observation.getInterpretation()` will return an object (not "null"), and the properties within the object will be null, so it won't throw a null exception